packages/react/src/components/Select/Select.tsx

```diff
@@ -169,9 +167,18 @@ const Select = React.forwardRef(function Select(
   const prefix = usePrefix();
   const { isFluid } = useContext(FormContext);
   const [isFocused, setIsFocused] = useState(false);
-  const [title, setTitle] = useState('');
   const selectInstanceId = useId();
 
+  // Convert children to an array of valid elements once
+  const validChildren = React.Children.toArray(children).filter(React.isValidElement);
+  // Find the default option based on the specified defaultValue
+  const defaultOption = validChildren.find(
+    (child) => child.props?.value === other?.defaultValue
+  );
+  // Use the default option's text if available; otherwise, fallback to the first option's text
+  const initialTitle = defaultOption?.props?.text || validChildren[0]?.props?.text || '';
```

Comment:
The TypeScript error Property 'text' does not exist on type '{}' is occurring because TypeScript doesn't recognize the text property on the props of child elements. 

```suggestion
 // Define an interface for the expected props
interface SelectItemProps {
  value: any;
  text: string;
}

// Use type narrowing instead of type assertion
const validChildren = React.Children.toArray(children)
  .filter((child): child is React.ReactElement<SelectItemProps> => 
    React.isValidElement<SelectItemProps>(child));

// Now TypeScript knows the shape of each child's props
const defaultOption = validChildren.find(
  (child) => child.props.value === other?.defaultValue
);
/ Use the default option's text if available; otherwise, fallback to the first option's text
  const initialTitle =
    defaultOption?.props?.text || validChildren[0]?.props?.text || '';
```

What are your views ?

Reply:
included this with minor changes

Reply:
the code has been changed to validate the child element and build error got resolved

Reply:
UPDATE: addressed the said issue, pushed the suggested change.